src/components/ThumbnailImage.js

```diff
@@ -54,8 +70,12 @@ class ThumbnailImage extends PureComponent {
         } else {
             thumbnailScreenHeight = Math.round(thumbnailScreenWidth * aspectRatio);
         }
+        return {width: thumbnailScreenWidth, height: thumbnailScreenHeight};
+    }
 
-        this.setState({thumbnailWidth: thumbnailScreenWidth, thumbnailHeight: Math.max(40, thumbnailScreenHeight)});
+    updateImageSize({width, height}) {
+        const {width: thumbnailWidth, height: thumbnailScreenHeight} = this.calculateThumbnailImageSize(width, height);
+        this.setState({thumbnailWidth, thumbnailHeight: Math.max(40, thumbnailScreenHeight)});
```

Comment:
Why this `thumbnailHeight: Math. max(40, thumbnailScreenHeight)}` and why don't do the same in the `constructor` i.e while setting the initial state?

Reply:
Check the diff. It was like this before. The comment states the reason. 
```
        // Width of the thumbnail works better as a constant than it does
        // a percentage of the screen width since it is relative to each screen
        // Note: Clamp minimum width 40px to support touch device 
```

Reply:
Added.